railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -244,11 +251,31 @@ def create_test_files
       end
 
       def create_tmp_files
-        build(:tmp)
+        build(:tmp) unless options[:api]
```

Comment:
Tmp might also be useful for temporary file processing, if you're not using a file service.